server/platform/services/sharedchannel/service.go

```diff
@@ -277,3 +277,18 @@ func (scs *Service) notifyClientsForSharedChannelUpdate(channel *model.Channel)
 	messageWs.Add("channel_id", channel.Id)
 	scs.app.Publish(messageWs)
 }
+
+// IsRemoteClusterDirectlyConnected checks if a remote cluster has a direct connection to the current server
+func (scs *Service) IsRemoteClusterDirectlyConnected(remoteId string) bool {
+	if remoteId == "" {
+		return true // Local server is always "directly connected"
+	}
+
+	// Check if the remote cluster exists, is online, and confirmed
+	rc, err := scs.server.GetStore().RemoteCluster().Get(remoteId, false)
+	if err != nil {
+		return false
```

Comment:
nit: do we want to log the error in case it's worth for an admin to check?

Reply:
We don't look to be doing it in other places for the same call, so I aligned to that.